### ISSUE TITLE
Kerberos (hive/presto) access documentation

### ISSUE THREAD

**Issue by parisni** [dropbox/PyHive#174]
Hi

Appently ( #47 #91 ) kerberized access is available.
However there is no example on how to use it in the documention.

That would be more than helpfull

Thanks

**Comment by ExpressGit** [comment]
Hi:
   
     where we can access the documention ,we need it

Thanks

**Comment by Dubrzr** [comment]
Looking at https://github.com/dropbox/PyHive/blob/master/pyhive/hive.py, here is how:

```
sudo apt-get install libsasl2-dev libsasl2-2 libsasl2-modules-gssapi-mit
```
```
pip install Pyhive sasl thrift_sasl 
```

```python
from pyhive import hive

conn = hive.Connection(host="<kerberos-server-host>", port=<port-of-hive-or-presto>, username="<user-kerberos>", auth='KERBEROS', kerberos_service_name=True)
```

**Comment by hellofuturecyj** [comment]
if keytab must be kinited before connection,that means i have to run pyhive code on hadoop cluster,right?

**Comment by Dubrzr** [comment]
Nope, you can kinit from a remote computer (through ports 88 tcp+udp) and then do remote pyhive. We do just that.

**Comment by samarth-goel-guavus** [comment]
> Nope, you can kinit from a remote computer (through ports 88 tcp+udp) and then do remote pyhive. We do just that.

@Dubrzr : Possible to give an example or point to any such link where you've done this?

**Comment by Dubrzr** [comment]
@samarth-goel-guavus 

1. Install kerberos on your own computer
2. Setup kerberos on your computer so that it connects to the remote kerberos server (/etc/krb5.conf)
3. Given a keytab file (provided by your kerberos administrator), you can authenticate your computer to the remote kerberos server using `kinit -kt your.keytab username@KERBEROS_REALM`
4. You can check that you have a valid kerberos ticket using `klist`
5. You can now launch pyhive with kerberos.

**Comment by dmueller1607** [comment]
Is there also a solution for Windows? The above example Hive connection seems not to work on my Windows client (kinit works fine for years on my pc). Hive server log sais: 
`java.lang.RuntimeException: org.apache.thrift.transport.TSaslTransportException: No data or no sasl data in the stream`

**Comment by abiwill** [comment]
> @samarth-goel-guavus
> 
> 1. Install kerberos on your own computer
> 2. Setup kerberos on your computer so that it connects to the remote kerberos server (/etc/krb5.conf)
> 3. Given a keytab file (provided by your kerberos administrator), you can authenticate your computer to the remote kerberos server using `kinit -kt your.keytab username@YOUR_KERBEROS_REALM`
> 4. You can check that you have a valid kerberos ticket using `klist`
> 5. You can now launch pyhive with kerberos.

How can we implement this in a docker setup.
How will a user add hive data source using the superset UI in this case.

**Comment by bkyryliuk** [comment]
@Dubrzr  thanks for providing examples here. it would be nice if you could add this information to the readme.

**Comment by lsgrep** [comment]
> @samarth-goel-guavus
> 
> 1. Install kerberos on your own computer
> 2. Setup kerberos on your computer so that it connects to the remote kerberos server (/etc/krb5.conf)
> 3. Given a keytab file (provided by your kerberos administrator), you can authenticate your computer to the remote kerberos server using `kinit -kt your.keytab username@YOUR_KERBEROS_REALM`
> 4. You can check that you have a valid kerberos ticket using `klist`
> 5. You can now launch pyhive with kerberos.

I did all of this, but it did not work.  However, creating a cache file & setting the `KRB5CCNAME` env variable did the trick for me.  

```
# you have to run this
cmd=f'kinit -kt {keytab_file} -c {ccache_file} {principal}'
os.environ['KRB5CCNAME'] = ccache_file
```

**Comment by roy-sounak** [comment]
@lsgrep Can you provide the full code snippet. It would be very helpful
